python/ray/train/base_trainer.py

```diff
@@ -379,6 +370,14 @@ class TrainTrainable(trainable_cls):
             def __repr__(self):
                 return super().__repr__()
 
+            @classmethod
+            def has_dataset(cls):
+                return has_dataset
+
+            @classmethod
+            def base_scaling_config(cls):
+                return scaling_config
+
```

Comment:
We should be using `self._merged_config` for both of these properties right? The merged config has the most up to date info if we are tuning over datasets or scaling config.

Reply:
What does it provide us that is different from the scaling_config?

Reply:
If scaling config is a hyperparameter that we are tuning over, then values of the scaling config after merging will be different than the one during Trainer instantiation. The merged config will contain the values that we are evaluating for this specific trial.

Reply:
Actually nvm, this is all happening prior to trainables being scheduled (and thus prior to merging the configs).

If we want to warn before any trial scheduling has occurred, then we have to be ok with the warning not being correct when tuning over the scaling config.

Reply:
+1, made the language more imprecise.